File: unity/Assets/Show/ShowType/OrangeShow.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace FiveButtons
{
    public class OrangeShow : Show
    {
        private const float SPEED = 0.01f;
        private const float SPEED_Y = SPEED * 1.8f;
        private const float RADIUS = 1.0f;
        private const float OFFSET_Y = 0.5f;
        private const float Y = 1.8f;
        
        protected override Color GetColor() => new Color(1.0f, 0.392f, 0.0f);

        protected override void Anim()
        {
            var t = _time * SPEED;
            var ty = _time * SPEED_Y;
            
            var x0 = Mathf.Cos(t) * RADIUS;
            var z0 = Mathf.Sin(t) * RADIUS;
            var y0 = Y + Mathf.Sin(ty) * OFFSET_Y;
            _balls[0].transform.localPosition = new Vector3(x0, y0, z0);

            var x1 = Mathf.Cos(t + Mathf.PI) * RADIUS;
            var z1 = Mathf.Sin(t + Mathf.PI) * RADIUS;
            var y1 = Y + Mathf.Sin(ty + Mathf.PI) * OFFSET_Y;
            _balls[1].transform.localPosition = new Vector3(x1, y1, z1);
        }
    }
}

File: babylonjs/src/Show/ShowType/orangeShow.ts
import {Color3, Vector3} from "@babylonjs/core";
import {Show} from "../show";

export class OrangeShow extends Show {
    private static SPEED: number = 0.055;
    private static SPEED_Y: number = OrangeShow.SPEED * 1.8;
    private static RADIUS: number = 2.0;
    private static OFFSET_Y: number = 0.5;
    private static Y: number = 1.8;

    protected ballCount(): number {
        return 2;
    }

    protected getColor(): Color3 {
        return new Color3(1, 0.45, 0.09);
    }

    protected anim(): void {
        const t = this._time * OrangeShow.SPEED;
        const ty = this._time * OrangeShow.SPEED_Y;

        const x0 = Math.cos(t) * OrangeShow.RADIUS;
        const z0 = Math.sin(t) * OrangeShow.RADIUS;
        const y0 = OrangeShow.Y + Math.sin(ty) * OrangeShow.OFFSET_Y;
        this._balls[0].position = new Vector3(x0, y0, z0);

        const x1 = Math.cos(t + Math.PI) * OrangeShow.RADIUS;
        const z1 = Math.sin(t + Math.PI) * OrangeShow.RADIUS;
        const y1 = OrangeShow.Y + Math.sin(ty + Math.PI) * OrangeShow.OFFSET_Y;
        this._balls[1].position = new Vector3(x1, y1, z1);
    }
}